plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -188,6 +195,38 @@ public FileHiveMetastore(NodeVersion nodeVersion, HdfsEnvironment hdfsEnvironmen
         }
     }
 
+    @Override
+    public synchronized long acquireTableExclusiveLock(
+            HiveIdentity identity,
+            AcidTransactionOwner transactionOwner,
+            String queryId,
+            String dbName,
+            String tableName)
+    {
+        SchemaTableName schemaTableName = new SchemaTableName(dbName, tableName);
+
+        Long existingLockId = tableLockIds.get(schemaTableName);
+        ReentrantLock lock = existingLockId == null ? new ReentrantLock() : tableLocks.get(existingLockId);
```

Comment:
you can use `computeIfAbsent` if you have maps

- table name -> lock
- lock id -> table name

that can also allow cleaning up locks so they aren't leaked